### PR TITLE
Improve instructions

### PR DESCRIPTION
**Pull request by afonsojramos**
The workshop is mostly ready, we just need to fix these TODOs.

Can you give some input on how to fix them?

### CODE REVIEW

File: workshop.ipynb [hunk 1]
```diff
@@ -9,8 +9,9 @@
     "In this workshop we will learn the basic functionalities of Git and how these can improve our team workflow.\n",
     "Most of this workshop will be conducted using a bash shell and as such some basic command line usage will also be exposed\n",
```
Comment (xfontes42) [inline workshop.ipynb:10]: "will also be exposed" doesn't sound good, I suggest "will also be **shown and explained**"

File: workshop.ipynb [hunk 2]
```diff
@@ -102,9 +165,19 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Adding files to our repository (15min)\n",
+    "#### 3.3 Command Breakdown\n",
+    "**mkdir** - Referenced [here](#2.1-Command-Breakdown)\n",
     "\n",
-    "Our repository is still empty so lets add all our files in one go (definetly not advisable but lets see why later)"
+    "**mv** - **M**o**v**e, the same idea as presented [here](#3.1-Command-Breakdown) but with some extra functionality. We attempt to move a file names \\*.js and although we have no file named exactly like that, all files that en in  \".js\" are moved. The reason why is a concept called regular expressions (look this up yourself its a really powerfull tool but this is not the place to go into detail, give us some feedback if you's like to see a orkhshop on this topic). The * symbol is reserved in regular expressions as a wildcard so any number of any carachater is represented by \\*, in this case, it means any file independently of its name will me selected to move as long as it ends in \".js\". The same aplies for the other terminations"
```
Comment (xfontes42) [inline workshop.ipynb:171]: "all files that **en**" -> **end**

Comment (xfontes42) [reply workshop.ipynb:171]: "orkshop" -> **workshop**

Comment (xfontes42) [reply workshop.ipynb:171]: "will **me**" -> **be**

File: workshop.ipynb [hunk 3]
```diff
@@ -135,6 +218,18 @@
     "git commit -m \"Initial Commit\""
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 4.2 Command Breakdown\n",
+    "**git reset** - using this command without arguments will alow us to reset the staging area mening the files we just added using. **git add** will be removed but the working tree will stay the same. ***WARNING***, the reset command provides a lot more functionality which at times may even deleted changed you've made to the working tree and subsequently all the working you still haven't commited, just make sure you know what exactly you are doing before using it.\n",
```
Comment (xfontes42) [inline workshop.ipynb:226]: "deleted" -> **delete**
"all the working" -> **work**

File: workshop.ipynb [hunk 4]
```diff
@@ -135,6 +218,18 @@
...
+    "\n",
+    "**git add** - Adding was covered [here](#4.1-Command-Breakdown) with an option, now using an argument we will just add the file that matches the parameter\n",
+    "\n",
+    "**git commit** - As the name says we are commiting to a change we made to the repository, once you commit a state of the repository it will be registered and you can always restore it or check differences to the current state. Commits can be deleted but its not something you do on a regular basis as it's usualy easier to fix something rather than redoing everything else. The \"m\" lets you add a message to your commit, these should usualy follow a protocol that lets you quicky understand what's beed added removed or modified with that commit"
```
Comment (xfontes42) [inline workshop.ipynb:230]: "once you commit **a** state of the repository **it** will" -> "once you commit the state of the repository will"

File: workshop.ipynb [hunk 5]
```diff
@@ -177,9 +281,25 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Branching and Merging for Parallel Work (15min)\n",
+    "#### 4.4 Command Breakdown\n",
+    "**echo** - This command is used to print information to the shell. When combined with the \">>\" pipe it is very usefull to dump information in files. So, the complete command writes the \"/lib\" which is redirected using \">>\" to the file named \".gitignore\", if the file doesn't exist it is created. The pipe \">\" also exists and the difference is that it appends while \">>\" overwrites the whole file if it already exists. Obviously, further on you will ignore files using the \">\" pipe.\n",
```
Comment (xfontes42) [inline workshop.ipynb:285]: "usefull" -> **useful**

File: workshop.ipynb [hunk 6]
```diff
@@ -177,9 +281,25 @@
...
+    "**git commit** - Referenced [here](#4.2-Command-Breakdown)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. Branching and Merging for Parallel Work (15min)\n",
+    "\n",
+    "Branching is the great feature of Git that will allow us to work in parallel in different or even similar sections of code with relative independance. This feature is highly useful for team-based projects, where someone needs to work on one section without interfeering with others' sections. To start with lets check what branches we currently have and create a new one both using the **branch** command. To edit our styles2.css file we want to use the new branch so we need to **checkout** taht one and use **branch** again just to make sure we are now in the correct branch."
```
Comment (xfontes42) [inline workshop.ipynb:302]: "taht" -> **that**

File: workshop.ipynb [hunk 7]
```diff
@@ -267,9 +419,21 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Remote repository (25min)\n",
+    "#### 5.4 Command Breakdown\n",
+    "**git merge** - Merge is again of the most powerful commands in git. As the name indicates it alows us to merge or bring together two different versions of our files. It wil try to automatically put together and resolve differences in the files from the parameter branch and into our current branch. There are several options you an explore and even specific strategies to use when merging, but it's never perfect and conflict can arise often as you've experienced. After the command is succsessfully completed the current branch will be up to date with all the changes made in the parameter branch.\n",
```
Comment (xfontes42) [inline workshop.ipynb:423]: "again of the most" -> "**one** of the most"

Comment (xfontes42) [reply workshop.ipynb:423]: "conflict" -> **conflicts**

Comment (xfontes42) [reply workshop.ipynb:423]: "an" -> **can**

File: workshop.ipynb [hunk 8]
```diff
@@ -267,9 +419,21 @@
...
+    "**git add** - Referenced [here](#4.2-Command-Breakdown)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 6. Remote repository (25min)\n",
+    "\n",
+    "Git works at a local level but it wouldn't mean much if we couldn't share out work with other people. So we want to make our repository available online. In this workshop we will make use of GitHub, an online, free-to-use platform for hosting git repositories. Start by creating an account (if you already have one that should be fine). <strong>Create an empty repository</strong> with a name of your choosing (one the home page use the green buttong on the left that sys new and do not fill any of the options that appear after the public, private selection). After completing this steps lets add this repository as the remote for the local one we have been working on.\n"
```
Comment (xfontes42) [inline workshop.ipynb:436]: "one" -> **on**
"buttong" -> **button**
"sys" -> **says**
"this" -> **these**

File: workshop.ipynb [hunk 9]
```diff
@@ -296,18 +470,31 @@
...
+    "cat ~/.ssh/id_rsa.pub #(or the save path you specified)",
+    "ssh-add ~/.ssh/id_rsa"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "#### 6.2 Command Breakdown\n",
+    "**ssh-keygen** - ssh-keygen is a command that comes pre-installed with most linux distributions and provides an user friendly way of generating several types of encryption keys. In this example we use the default key pair, RSA, and recommend using the default save path and you may choose to pretect the key with a password. Encryption keys are a complex subject that veer too much from the topic of this workshop so we suggest you look these up by yourself.\n",
```
Comment (xfontes42) [inline workshop.ipynb:485]: "pretect" -> **protect**

File: workshop.ipynb [hunk 10]
```diff
@@ -296,18 +470,31 @@
...
     "\n",
-    "Using the cat command we can now copy the generated key and copy it into the github configuration in the \"SSH and GPG Keys\" section in settings. We should now be able to freely access our repository!\n"
+    "**cat** - Con**cat**tenate, is a very flexible command that we are using to diplay the contents of the file that contains the public part of the key pair we just generated. It's very important to remember that we should only ever share the public pieces of any key as having access to the private one would mean we could be impersonated by the holder of that key. Most key files are names \".pub\" to signal they are the public part of the key.",
```
Comment (xfontes42) [inline workshop.ipynb:487]: "diplay" -> **display**
"names" -> **named**

File: workshop.ipynb [hunk 11]
```diff
@@ -343,6 +540,27 @@
...
+    "**git log** - Referenced [here](#4.3-Command-Breakdown)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The process of fetching and merging can be reduced to the **git pull** command which will perform both of these commands and try to merge changes automatically.\n\n",
+    "If you want to improve this workshop, or just to check it out on GitHub (you can also leave a star ;)), you can do it [here](https://github.com/acmfeup/git-workshop).\n",
+    "If you enjoyed this workshop and would like to help us creating more projects, workshops, conferences and others, please do not hesitate in contacting us through one of our various social media platforms, or emailing us @ neacm@fe.up.pt."
```
Comment (msramalho) [inline workshop.ipynb:562]: maybe a mailto: link on the email address

**Review by msramalho — APPROVED** [review]
Quick comments: Great idea to use jupyterhub with a bash kernel so there are no compatibility issues to handle, although I think it might also be important to leave some installations instructions of git for windows and linux, maybe at the end.

Other than that, looks good and I would add links to either documentation or examples for each shell command use (mkdir, cat, ...)

**Comment by msramalho** [comment]
Also, following @xfontes42 typo corrections I'd recommend using something like [jupyter spellchecker](https://qingkaikong.blogspot.com/2018/09/spell-checking-in-jupyter-notebook.html) to catch them while developing the notebooks.

**Comment by msramalho** [comment]
Also, following @xfontes42 typo corrections I'd recommend using something like [jupyter spellchecker](https://qingkaikong.blogspot.com/2018/09/spell-checking-in-jupyter-notebook.html) to catch them while developing the notebooks.

**Comment by antonioalmeida** [comment]
@msramalho @afonsojramos @xfontes42 @edramos-97  Please take a loot at #3 before further reviews.

**Pull request closed without merging**